layers/bfd.go

```diff
@@ -0,0 +1,507 @@
+// Copyright 2017 Google, Inc. All rights reserved.
+//
+// Use of this source code is governed by a BSD-style license
+// that can be found in the LICENSE file in the root of the source
+// tree.
+//
+//******************************************************************************
+
+package layers
+
+import (
+	"encoding/binary"
+	"errors"
+
+	"github.com/google/gopacket"
+)
+
+//******************************************************************************
+//
+// Bidirectional Forwarding Detection (BFD) Decoding Layer
```

Comment:
Consider moving some of this (probably everything except the stuff copied directly from the RFC?) into the godoc comment for the BFD struct.